src/main/groovy/com/google/appengine/task/endpoints/WebXmlProcessing.groovy

```diff
@@ -36,7 +36,28 @@ class WebXmlProcessing {
     }
 
     private static NodeChild getSystemServiceServlet(GPathResult root) {
-        def servlets = root."servlet".findAll { it."servlet-class".text().trim() == "com.google.api.server.spi.SystemServiceServlet" ? it : null }
+        def servlets = root."servlet".findAll { servlet ->
```

Comment:
For example I'm using custom overridden implementation of SystemServiceServlet
```xml
	<servlet>
		<servlet-name>SystemServiceServlet</servlet-name>
		<servlet-class>com.severn.common.http.WrappedSystemServiceServlet</servlet-class>
		<init-param>
			<param-name>services</param-name>
			<param-value></param-value>
		</init-param>
        <init-param>
            <param-name>logRequest</param-name>
            <param-value>false</param-value>
        </init-param>
        <init-param>
            <param-name>logResponse</param-name>
            <param-value>true</param-value>
        </init-param>
        <init-param>
            <param-name>endpointsPath</param-name>
            <param-value>/_ah/spi/com.severn</param-value>
        </init-param>
	</servlet>
    <servlet-mapping>
        <servlet-name>SystemServiceServlet</servlet-name>
        <url-pattern>/_ah/spi/*</url-pattern>
    </servlet-mapping>
```
If we stick only to servlet class, it won't allow to find endpoint classes and fails to generate discovery docs in cases like above. Servlet name is more flexible, however, is not perfect.
Related TODOs are: 
1. Find servlet name by its mapping `/_ah/spi/*`, and then find servlet declaration, then get `services` init param.
2. Implement a search of endpoint services by @Api annotations like maven plugin does and put them to the services param  

Agree, it should go in another PR.